packages/compartment-mapper/src/bundle.js

```diff
@@ -298,6 +344,226 @@ ${''.concat(...modules.map(m => m.bundlerKit.getFunctorCall()))}\
   return bundle;
 };
 
+function wrapFunctorInPrecompiledModule(functorSrc, compartmentName) {
+  const wrappedSrc = `() => (function(){
+  with (this.scopeTerminator) {
+  with (this.globalThis) {
+    return function() {
+      'use strict';
+      return (
+${functorSrc}
+      );
+    };
+  }
+  }
+}).call(getEvalKitForCompartment(${JSON.stringify(compartmentName)}))()`;
+  return wrappedSrc;
+}
+
+// This function is serialized and references variables from its destination scope.
+
+function renderFunctorTable(functorTable) {
+  const entries = Object.entries(functorTable);
+  const lines = entries.map(
+    ([key, value]) => `${JSON.stringify(key)}: ${value}`,
+  );
+  return `{\n${lines.map(line => `  ${line}`).join(',\n')}\n};`;
+}
+
+/**
+ * @param {CompartmentMapDescriptor} compartmentMap
+ * @param {Sources} sources
+ * @returns {Promise<string>}
+ */
+export const makeSecureBundleFromAppContainer = async (
```

Comment:
On the principle of importing-what-you-paid-for, this could be factored into `secure-bundle.js`.